Dockerfile

```diff
@@ -17,79 +26,88 @@ ENV PATH /opt/conda/bin:$PATH
     curl -L "https://github.com/krallin/tini/releases/download/v${TINI_VERSION}/tini_${TINI_VERSION}.deb" > tini.deb && \
     dpkg -i tini.deb && \
     rm tini.deb && \
-    apt-get clean
+    apt-get clean && \
+    rm -rf /var/lib/apt/lists/*
+
 
 # Install a few dependencies for iCommands, text editing, and monitoring instances
-RUN apt-get update && apt-get install -y lsb-release apt-transport-https curl supervisor nginx gnupg2 libfuse2 gcc less nodejs software-properties-common apt-utils glances htop vim emacs nano 
+RUN apt-get update && \
+    apt-get install -y lsb-release apt-transport-https curl gnupg2 libfuse2 gcc less nodejs software-properties-common apt-utils glances htop nano  && \
+    apt-get clean && \
+    rm -rf /var/lib/apt/lists/*
+
+RUN apt-get update && \
+    apt-get install -y vim-nox && \
+    apt-get clean && \
+    rm -rf /var/lib/apt/lists/*
+
+RUN apt-get update && \
+    apt-get install -y emacs-nox && \
+    apt-get clean && \
+    rm -rf /var/lib/apt/lists/*
 
 RUN wget -qO - https://packages.irods.org/irods-signing-key.asc | apt-key add - && \
     echo "deb [arch=amd64] https://packages.irods.org/apt/ bionic main" > /etc/apt/sources.list.d/renci-irods.list && \
     apt-get update && \
-    wget -c \
+    wget -q -c \
     http://security.ubuntu.com/ubuntu/pool/main/p/python-urllib3/python-urllib3_1.22-1ubuntu0.18.04.2_all.deb \
     http://security.ubuntu.com/ubuntu/pool/main/r/requests/python-requests_2.18.4-2ubuntu0.1_all.deb \
-    http://security.ubuntu.com/ubuntu/pool/main/o/openssl1.0/libssl1.0.0_1.0.2n-1ubuntu5.6_amd64.deb && \
+    http://security.ubuntu.com/ubuntu/pool/main/o/openssl1.0/libssl1.0.0_1.0.2n-1ubuntu5.7_amd64.deb && \
     apt install -y \
     ./python-urllib3_1.22-1ubuntu0.18.04.2_all.deb \
     ./python-requests_2.18.4-2ubuntu0.1_all.deb \
-    ./libssl1.0.0_1.0.2n-1ubuntu5.6_amd64.deb && \
+    ./libssl1.0.0_1.0.2n-1ubuntu5.7_amd64.deb && \
     rm -rf \
     ./python-urllib3_1.22-1ubuntu0.18.04.2_all.deb \
     ./python-requests_2.18.4-2ubuntu0.1_all.deb \
-    ./libssl1.0.0_1.0.2n-1ubuntu5.6_amd64.deb
-
-RUN apt install -y irods-icommands
+    ./libssl1.0.0_1.0.2n-1ubuntu5.7_amd64.deb && \
+    apt install -y irods-icommands && \
+    apt-get clean && \
+    rm -rf /var/lib/apt/lists/*
 
 # Install Go
-RUN wget -c https://dl.google.com/go/go1.16.4.linux-amd64.tar.gz -O - | sudo tar -xz -C /usr/local
+RUN wget -q -c https://dl.google.com/go/go1.16.4.linux-amd64.tar.gz -O - | sudo tar -xz -C /usr/local
 
 # install ttyd
-RUN apt-get update && apt-get install -y libjson-c-dev libwebsockets-dev cmake
-RUN git clone https://github.com/tsl0922/ttyd.git && \
-    cd ttyd && mkdir build && cd build && \
-    cmake .. && \
-    make && make install
+RUN curl -L "https://github.com/tsl0922/ttyd/releases/download/1.6.3/ttyd.x86_64" > ttyd && \
+    chmod a+x ttyd && \
+    mv ttyd /usr/local/bin/ttyd
```

Comment:
This is an easier and more space-efficient way to install ttyd, just grabbing a prebuilt binary instead of building it.